_posts/2016-03-31-Adding-TypeScript-to-Ionic-Framework.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+author: jan_de_wilde
+title: 'Adding TypeScript to Ionic Framework'
+image: /img/ionic-and-typescript.jpg
+tags: [TypeScript,Ionic]
+category: Ionic
+comments: true
+---
+
+### Ionic and TypeScript sitting in a tree
+
+So, [TypeScript](http://www.typescriptlang.org/) is the all new thing that allows you to use features from ES6 (or ES2015), ES7 and beyond. Say goodbye to loosely typed variables and say hello to modules, classes, interfaces and so much more.
+
+In order to use TypeScript in an Ionic Framework project there are a few small things you need to do to get things running.
+
+## 1. Install and configure the gulp package
+
+* Install the **gulp-tsc package** and save it to the development dependencies in `package.json`
+
+   {% highlight text %}npm install gulp-tsc --save-dev{% endhighlight %}
+
+* Next, require the package in your **gulpfile.js** like so
+
+   {% highlight text %}var typescript = require('gulp-tsc');{% endhighlight %}
+
+* Add the following line to the paths object: `ts: ['./src/*.ts', './src/**/*.ts']`. You may have noticed two things here: I'm using a **src folder** to place all my TypeScript files in which means I'm **not using the www folder** that Ionic provides by default. Next to that I'm also targeting subfolders in that folder because I'm bundling my logic based on AngularJS modules. You can read more about structuring an AngularJS project in the [John Papa AngularJS Style Guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#application-structure-lift-principle).
```

Comment:
"I'm using a folder to place all my Typescript files in"

"You may have noticed two things here: All my TypeScript files are in a **src folder** which means..."